handlers/api.go

```diff
@@ -212,9 +212,37 @@ func ApiEthStoreDay(w http.ResponseWriter, r *http.Request) {
 
 	var err error
 	var rows *sql.Rows
-	query := `
+
+	var day int64
+
+	vars := mux.Vars(r)
+	if vars["day"] == "latest" {
+		err = db.ReaderDb.Get(&day, `
+			SELECT 
+				MAX(day)
+			FROM eth_store_stats e
+			WHERE validator = -1 `)
+		if err != nil {
+			logger.Errorf("error retrieving latest day: %v", err)
+			sendErrorResponse(w, r.URL.String(), "could not retrieve db results")
+			return
+		}
+	} else {
+		day, err = strconv.ParseInt(vars["day"], 10, 64)
+		if err != nil {
+			sendErrorResponse(w, r.URL.String(), "invalid day provided")
+			return
+		}
+	}
+
+	dayStart := utils.DayToTime(day).Unix()
+	dayEnd := utils.DayToTime(day + 1).Unix()
+
+	rows, err = db.ReaderDb.Query(`
 		SELECT 
-			day, 
+			day,
+			to_timestamp($1) as day_start,
+			to_timestamp($2) as day_end,
```

Comment:
This is misusing Postgres to marry the query result with your `dayStart` / `dayEnd` calculations from above, right?

I would not do that. Consider the same approach as commented below in line 1540. That way you would even get back to a single db query.

Reply:
I agree, I just thought it was fine since `/api/v1/epoch/{epoch}` does the exact same thing.
Since I rewrote the query result response code to work for everything I also fixed it for `/api/v1/epoch/{epoch}`.

I didn't search through all APIs to see if this could be improved anywhere else since I don't know how to do so efficiently.